File: Week 03/id_441/assign-cookies.js
// https://leetcode-cn.com/problems/assign-cookies/
var findContentChildren = function (g, s) {
    let res = 0;
    s = s.sort((a, b) => a - b)

    g = g.sort((a, b) => a - b)

    g.forEach(item => {
        while (s.length) {
            if (item <= s.shift()) {
                res++
                break
            }
        }
    })
    return res
}

var findContentChildren = function (g, s) {
    s = s.sort((a, b) => a - b)

    g = g.sort((a, b) => a - b)

    let g_i = 0;
    let s_i = 0;
    let count = 0;
    while (g_i < g.length && s_i < s.length) {
        if (g[g_i] <= s[s_i]) {
            g_i++;
            s_i++;
            count += 1
        } else {
            s_i++
        }
    }

    return count
}

File: Week 03/id_011/assign-cookies.go
package algorithm00401

import "sort"

func findContentChildren(g []int, s []int) int {
	sort.Sort(sort.Reverse(sort.IntSlice(g)))
	sort.Sort(sort.Reverse(sort.IntSlice(s)))

	var res int
	gi, si := 0, 0
	for gi < len(g) && si < len(s) {
		if s[si] >= g[gi] {
			res++
			gi++
			si++
		} else {
			gi++
		}
	}
	return res
}
